src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -74,9 +83,29 @@ public static boolean unlock(final Device dev) {
                     break;
                 }
             }
-            n--;
+            n++;
         }
-        return false;
+        if (!isUnlocked) {
```

Comment:
This is the new heuristic, and the number of spins in this heuristic are always random.